flytekit/image_spec/default_builder.py

```diff
@@ -0,0 +1,245 @@
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+import warnings
+from pathlib import Path
+from string import Template
+from typing import ClassVar
+
+import click
+
+from flytekit.image_spec.image_spec import (
+    _F_IMG_ID,
+    ImageSpec,
+    ImageSpecBuilder,
+)
+from flytekit.tools.ignore import DockerIgnore, GitIgnore, IgnoreGroup, StandardIgnore
+
+PYTHON_INSTALL_COMMAND_TEMPLATE = Template("""\
+RUN --mount=type=cache,sharing=locked,mode=0777,target=/root/.cache/uv,id=uv \
+    --mount=from=uv,source=/uv,target=/usr/bin/uv \
+    --mount=type=bind,target=requirements.txt,src=requirements.txt \
+    /usr/bin/uv \
+    pip install --python /root/micromamba/envs/dev/bin/python $PIP_EXTRA \
+    --requirement requirements.txt
+""")
+
+APT_INSTALL_COMMAND_TEMPLATE = Template(
+    """\
+RUN --mount=type=cache,sharing=locked,mode=0777,target=/var/cache/apt,id=apt \
+    apt-get update && apt-get install -y --no-install-recommends \
+    $APT_PACKAGES
+"""
+)
+
+DOCKER_FILE_TEMPLATE = Template(
+    """\
+#syntax=docker/dockerfile:1.5
+FROM ghcr.io/astral-sh/uv:0.2.13 as uv
+FROM mambaorg/micromamba:1.5.8-bookworm-slim as micromamba
+
+FROM $BASE_IMAGE
+
+USER root
+$APT_INSTALL_COMMAND
+RUN update-ca-certificates
+
+RUN id -u flytekit || useradd --create-home --shell /bin/bash flytekit
+RUN chown -R flytekit /root && chown -R flytekit /home
+
+RUN --mount=type=cache,sharing=locked,mode=0777,target=/root/micromamba/pkgs,\
+id=micromamba \
+    --mount=from=micromamba,source=/usr/bin/micromamba,target=/usr/bin/micromamba \
+    /usr/bin/micromamba create -n dev -c conda-forge $CONDA_CHANNELS \
+    python=$PYTHON_VERSION $CONDA_PACKAGES
```

Comment:
This mounts the executable `micromamba` from `mambaorg/micromamba:1.5.8-bookworm-slim`, which means the executable does not end up in the base image. This results in a smaller final image.

I do the same thing for `uv`.